common/types.go

```diff
@@ -186,6 +186,42 @@ func (h Hash) Value() (driver.Value, error) {
 	return h[:], nil
 }
 
+// GetLastNumBitsFromHash returns the last num bits of the hash as int64
+func (h Hash) GetLastNumBitsFromHash(num int) int64 {
+	if num > 64 {
+		return 0
```

Comment:
Returning -1 is probably the desired behavior.

Reply:
or panic

Reply:
panic it is.